src/main/java/net/minecraftforge/common/MapGenStructureManager.java

```diff
@@ -152,6 +152,8 @@ public void generate(World worldIn, int x, int z, ChunkPrimer primer)
         /**
          * Collect the list of creatures of the given type that should spawn within the structures bounding box.
          * <p>
+         * If this method is called repeatedly you should always return exactly the same (equals) objects otherwise the spawning will not work.
```

Comment:
why? is the structure not allowed to vary spawns based on e.g. time of day?

Reply:
@williewillus The worldentityspawner calls this method twice.
1) to get possible spawns
2) to check if the selected spawn entry is allowed to spawn here
The entry selected after the first call has to be contained (list#contains) in the result of the second call otherwise the entity won't spawn.

But you can vary this over time of course.
I will try to clarify this comment